libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -45,129 +47,139 @@ interface IRushXCommandLineArguments {
 }
 
 export class RushXCommandLine {
-  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
-    RushXCommandLine._launchRushXInternal(launcherVersion, { isManaged });
+  public static launchRushX(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+    try {
+      const args: IRushXCommandLineArguments = this._getCommandLineArguments();
```

Comment:
args and configuration were moved up to here to support calling eventhooks efficiently.